### ISSUE TITLE
Undefined method `exitstatus' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by fastlanebot** [fastlane/fastlane#2077]
<table>
      <tr>
        <td>
          <img src='https://avatars0.githubusercontent.com/u/8449621?v=3&s=70' width='35'>
        </td>
        <td>
          Original issue by @VictorJava - Imported from <a href='https://github.com/fastlane/screengrab/issues/84'>fastlane/screengrab#84</a>
        </td>
      </tr>
    </table>

Hi guys, 

I trying start screengrab in Ubuntu with Android project.

My way: call screengrab in terminal -> put package -> see error with exitstatus.

After sudo gem install fastlane
I sow error like use ruby >= 2.2.2
So, ruby was updated up to 2.3.
But problem hasn`t gone.

Please, see log below error message
 while running screengrab
 and fastlane.
Does fastlane need install for correct work with screengrab?
Could you help me?

~ $ screengrab
.......
$ adb devices -l
[12:55:24]: ▸ List of devices attached
[12:55:24]: ▸ 08a07e0c43aab722       device usb:1-2 product:hammerhead model:Nexus_5 device:hammerhead
-------------------------------------------------------------------------------------------
😨  An error occured. Please enable crash reports using `fastlane enable_crash_reporting`.
👍  This makes resolving issues much easier and helps improve fastlane.
🔒  The reports will be stored securely on getsentry.com.
🙊  More information about privacy: https://github.com/fastlane/fastlane/releases/tag/1.33.3
-------------------------------------------------------------------------------------------
/var/lib/gems/2.1.0/gems/fastlane_core-0.36.8/lib/fastlane_core/command_executor.rb:79:in `execute': [!] undefined method `exitstatus' for nil:NilClass (NoMethodError)
	from /var/lib/gems/2.1.0/gems/screengrab-0.2.1/lib/screengrab/runner.rb:80:in `select_device'
	from /var/lib/gems/2.1.0/gems/screengrab-0.2.1/lib/screengrab/runner.rb:52:in `run'
	from /var/lib/gems/2.1.0/gems/screengrab-0.2.1/bin/screengrab:39:in `block (2 levels) in run'
	from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /var/lib/gems/2.1.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /var/lib/gems/2.1.0/gems/screengrab-0.2.1/bin/screengrab:56:in `run'
	from /var/lib/gems/2.1.0/gems/screengrab-0.2.1/bin/screengrab:62:in `<top (required)>'
	from /usr/local/bin/screengrab:23:in `load'
	from /usr/local/bin/screengrab:23:in `<main>'
~ $ ruby -v
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux-gnu]
~ $ sudo gem install fastlane
Building native extensions.  This could take a while...
ERROR:  Error installing fastlane:
	ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.3.0/gems/unf_ext-0.0.7.2/ext/unf_ext
/usr/bin/ruby2.3 -r ./siteconf20160301-30575-gfbvtm.rb extconf.rb
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h

extconf failed, exit code 1

Gem files will remain installed in /var/lib/gems/2.3.0/gems/unf_ext-0.0.7.2 for inspection.
Results logged to /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/unf_ext-0.0.7.2/gem_make.out
~ $ 

Thanks for your help.

**Comment by chteuchteu** [comment]
I'm currently experiencing the same issue.
Screengrab correctly prints `adb devices -l` output, but then seems to fail reading it:

`/var/lib/gems/2.0.0/gems/fastlane_core-0.37.0/lib/fastlane_core/command_executor.rb:79:in `execute': [!] undefined method `exitstatus' for nil:NilClass (NoMethodError)`

Running `adb devices -l ; echo "result code: $?"` outputs `return code: 0`.

**Comment by lvh1** [comment]
Any updates on this?
I'm also running into this issue, also on Ubuntu 15.10.

**Comment by lvh1** [comment]
Thanks for the very fast response @i2amsam 
I updated to the gems to the latest version, but unfortunately the problem persists :disappointed:.
```
😨  An error occured. Please enable crash reports using `fastlane enable_crash_reporting`.
👍  This makes resolving issues much easier and helps improve fastlane.
🔒  The reports will be stored securely on getsentry.com.
🙊  More information about privacy: https://github.com/fastlane/fastlane/releases/tag/1.33.3
-------------------------------------------------------------------------------------------
/var/lib/gems/2.1.0/gems/fastlane_core-0.39.0/lib/fastlane_core/command_executor.rb:79:in `execute': [!] undefined method `exitstatus' for nil:NilClass (NoMethodError)
        from /var/lib/gems/2.1.0/gems/screengrab-0.3.1/lib/screengrab/runner.rb:80:in `select_device'
        from /var/lib/gems/2.1.0/gems/screengrab-0.3.1/lib/screengrab/runner.rb:52:in `run'
        from /var/lib/gems/2.1.0/gems/screengrab-0.3.1/bin/screengrab:39:in `block (2 levels) in run'
        from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
        from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
        from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
        from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
        from /var/lib/gems/2.1.0/gems/fastlane_core-0.39.0/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
        from /var/lib/gems/2.1.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
        from /var/lib/gems/2.1.0/gems/screengrab-0.3.1/bin/screengrab:56:in `run'
        from /var/lib/gems/2.1.0/gems/screengrab-0.3.1/bin/screengrab:62:in `<top (required)>'
        from /usr/local/bin/screengrab:22:in `load'
        from /usr/local/bin/screengrab:22:in `<main>'
```

**Comment by i2amsam** [comment]
@lvh1 could you try this?

```
gem install fastlane_core -v 0.38.0
gem uninstall fastlane_core -v 0.39.0
```
and try again?  As tracked in #3821 I pushed a second update to revert the change while I track down some test flakiness, but the fix in `0.38.0` may solve this problem, it would be awesome if you could verify.

**Comment by lvh1** [comment]
That did seem to fix it. Thanks a lot!
